### PR TITLE
Use AppVeyor API tokens specified per user

### LINKED ISSUE #9: Limit authorised users of AppVeyor commands to some defined set
Right now, anybody who can talk to the bot can issue any of the commands in this script. In order to prevent malicious activity and to _save us time_ when someone makes a mistake, we should only allow authorised users to issue these commands.

I would expect the bot to respond with something similar to:

```
SomeGuy> @hubot: start build something
hubot>   @SomeGuy you need to be authorised to issue that command.
```

### PR DESCRIPTION
**Pull request by Codesleuth**
Fixes #9 by requiring users to set their own AppVeyor API token to be able to use commands.

Set your token with `@hubot: set token lkjowqen23nio32`

### CODE REVIEW

File: src/hubot.d.ts [hunk 1]
```diff
@@ -60,8 +60,8 @@ declare module "hubot" {
     users(): { [id: string]: User; };
     userForName(name: string): User;
     userForId(id: string, options: any): User;
-    get(key: string): string;
-    set(key: string, value: string): Brain;
+    get(key: string): any;
```
Comment (hawx) [inline src/hubot.d.ts]: `any`?

Comment (Codesleuth) [inline src/hubot.d.ts]: The brain can store anything, and the redis plugin does a `JSON.stringify` anyway. I do need to test this on a real redis brain.

File: src/test/build.test.ts [hunk 2]
```diff
@@ -191,4 +199,44 @@ test('finbot > starts a build > handles non-200 response', (t) => {
 
   // assert
   sinon.assert.calledWith(replyStub, `Could not start build. Got status code 403`);
+});
+
+test('finbot > start build > when no token set', (t) => {
+  // arrange
```
Comment (Blundell89) [inline src/test/build.test.ts]: WHY

**Pull request merged**